src/gs-image-tile.c

```diff
@@ -105,16 +105,67 @@ app_state_changed (GsApp *app, GParamSpec *pspec, GsImageTile *tile)
 	g_idle_add (app_state_changed_idle, g_object_ref (tile));
 }
 
+typedef struct _AppTileCSSChangedUserData
+{
+	GsApp *app;
+	GsAppTile *tile;
+} AppTileCSSChangedUserData;
+
+static void
+app_tile_css_changed_user_data_destroy (AppTileCSSChangedUserData *data)
+{
+	g_clear_object (&data->app);
+	g_clear_object (&data->tile);
+
+	g_free (data);
+}
+
+static AppTileCSSChangedUserData *
+app_tile_css_changed_user_data_new (GsApp *app, GsAppTile *tile)
+{
+	AppTileCSSChangedUserData *data = g_new0 (AppTileCSSChangedUserData, 1);
+	data->app = g_object_ref (app);
+	data->tile = g_object_ref (tile);
+	return data;
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(AppTileCSSChangedUserData, app_tile_css_changed_user_data_destroy);
+
+static gboolean
+app_tile_css_changed_idle (gpointer user_data)
+{
+	g_autoptr(AppTileCSSChangedUserData) data = user_data;
```

Comment:
Just call the free function after setting the CSS -- I think it's clearer for a simple struct like this.